src/python/pants/option/parser.py

```diff
@@ -503,6 +505,38 @@ def error(
         if not is_enum and member_type not in self._allowed_member_types:
             error(InvalidMemberType, member_type=member_type.__name__)
 
+        # check type of default value
+        default_value = kwargs.get("default")
+        if default_value is not None:
+            if isinstance(default_value, str) and type_arg != str:
+                # attempt to parse default value, for correctness..
+                # custom function types will have to implement their own validation
+                default_value = self.to_value_type(default_value, type_arg, member_type, "")
```

Comment:
It looks like the `dest` argument to `to_value_type` is unused...? Could probably be removed.

Reply:
it is used in the ParseError exception message.. but since we do not re-use that message here, but rather makes a new one, in this case dest is unused.